### DISCUSSION TITLE
Front end component only rendering after loader returns data

### DISCUSSION THREAD

**Discussion by treasurely** [remix-run/remix#3739] [Q&A]
Hi!  I'm trying to get a simple example going & I'm stuck getting the following to work:

My app structure includes something like this:

**app/routes/accounts.jsx**
**app/routes/accounts/index.jsx**

When I set the loader to only return after 5 seconds/some long delay, the entire component doesn't render until the loader function returns.  

I feel like there's something I'm missing on the set up here -- I'm looking to get accounts.jsx (parent route) to render, and then after the data loads the accounts table renders.  For the 5 seconds in between, something like a loading spinner/state would show in accounts/index.jsx.

From the accounts/index.jsx route:

```
export const loader = async () => {
  await new Promise((res) => {
    setTimeout(() => res(""), 5000);
  });

  const { data: accounts, error } = await supabase.from("accounts").select("*");
  return json(accounts);
};

export default function AccountsTable() {
  let data = useLoaderData();

  return (
    <div>
      <h3>Accounts Index</h3>
      {**Accounts Table**}
    </div>
  );
}
```

Thanks!

**Comment by kiliman** [comment]
Exactly. That's by design. Remix will call all loaders in parallel and will not render the route until all the data has been returned.

With that being said, they are working on a new _deferred_ feature, that will let you send data as well as one or more promises. This way slow data can be deferred. The data will stream to the client as it comes and you can then render it (it will render a fallback while waiting). This should be out hopefully by end of summer.

In the meantime, you can simply not return the *slow* data from the loader, then `useFetcher` to get it from the client. This way your routes will render quickly, but you'll have to show a spinner or something while the slow client-side fetch is happening.

**Reply by sergiodxa** [reply]
You can also use the useTransition hook to know if there's a pending navigation happening and render a spinner.
